### PR TITLE
Proxy relax sni check

### PR DESCRIPTION
**Pull request by khanova**
## Problem

Using the same domain name () for serverless driver can help with connection caching.
https://github.com/neondatabase/neon/issues/6290

## Summary of changes

Relax SNI check.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: proxy/src/serverless/sql_over_http.rs [hunk 1]
```diff
@@ -171,7 +172,8 @@ fn get_conn_info(
         .and_then(|h| h.to_str().ok())
         .and_then(|h| h.split(':').next());
 
-    if hostname != sni_hostname {
+    // sni_hostname has to be either the same as hostname or the one used in serverless driver.
+    if sni_hostname != SERVERLESS_DRIVER_SNI_HOSTNAME || sni_hostname != hostname {
         return Err(anyhow::anyhow!("mismatched SNI hostname and hostname"));
     } else if let Some(h) = host_header {
         if h != hostname {
```
Comment (conradludgate) [inline proxy/src/serverless/sql_over_http.rs:179]: same needs to happen here, afaiu

Comment (conradludgate) [reply proxy/src/serverless/sql_over_http.rs:179]: ah, and you need to make this an `&&`, not an or 😅

Comment (khanova) [reply proxy/src/serverless/sql_over_http.rs:179]: Oh, thanks!

Comment (conradludgate) [reply proxy/src/serverless/sql_over_http.rs:179]: the host header validation still needs to be changed

**Comment by conradludgate** [comment]
can we add a python test for this? a very simple `select 1` should do

File: proxy/src/serverless/sql_over_http.rs (hunk 1 above)
Comment (conradludgate) [reply proxy/src/serverless/sql_over_http.rs:179]: see
```
curl -k -v 'https://api.cloud.chloroform.aws.neon.build/sql' \
  -H 'Neon-Connection-String: postgresql://***@***.cloud.chloroform.aws.neon.build/neondb' \
  -H 'Content-Type: application/json' \
  -H 'Neon-Raw-Text-Output: true' \
  -H 'Neon-Array-Mode: false' \
  --data '{
    "query":"SHOW server_version_num",
    "params":[]
  }' | jq

> POST /sql HTTP/1.1
> Host: api.cloud.chloroform.aws.neon.build
> User-Agent: curl/8.1.2
> Accept: */*
> Neon-Connection-String: postgresql://***@***.cloud.chloroform.aws.neon.build/neondb
> Content-Type: application/json
> Neon-Raw-Text-Output: true
> Neon-Array-Mode: false
> Content-Length: 60
```

Comment (khanova) [reply proxy/src/serverless/sql_over_http.rs]: Thanks a lot! Updated and added test

**Review by conradludgate — APPROVED** [review]

**Review by conradludgate — APPROVED** [review]

**Pull request merged**